statsforecast/models.py

```diff
@@ -17,55 +19,141 @@
 from .arima import auto_arima_f, forecast_arima, fitted_arima
 from .ets import ets_f, forecast_ets
 
-# Cell
-def auto_arima(X: np.ndarray, # time series
-               h: int, # forecasting horizon
-               future_xreg=None, #future regressors
-               fitted: bool=False, # fitted values
-               season_length: int=1, # season_length
-               approximation: bool=False, # approximation
-               level: Optional[Tuple[int]] = None # level
-    ) -> np.ndarray:
-    y = X[:, 0] if X.ndim == 2 else X
-    xreg = X[:, 1:] if (X.ndim == 2 and X.shape[1] > 1) else None
-    with np.errstate(invalid='ignore'):
-        mod = auto_arima_f(
-            y,
-            xreg=xreg,
-            period=season_length,
-            approximation=approximation,
-            allowmean=False, allowdrift=False #not implemented yet
-        )
-    fcst = forecast_arima(mod, h, xreg=future_xreg, level=level)
-    mean = fcst['mean']
-    if fitted:
-        return {'mean': mean, 'fitted': fitted_arima(mod)}
-    if level is None:
+# %% ../nbs/models.ipynb 6
+class _TS:
+    
+    def new(self):
+        b = type(self).__new__(type(self))
+        b.__dict__.update(self.__dict__)
+        return b
+
+# %% ../nbs/models.ipynb 8
+class AutoARIMA(_TS):
+    
+    def __init__(
+            self, 
+            season_length: int = 1, # Number of observations per cycle
+            approximation: bool = False, #
+        ):
+        self.season_length = season_length
+        self.approximation = approximation
+        
+    def __repr__(self):
+        return f'AutoARIMA'
+    
+    def fit(
+            self, 
+            y: np.ndarray, # time series
+            X: Optional[np.ndarray] = None, # exogenous regressors
+        ):
+        with np.errstate(invalid='ignore'):
+            self.model_ = auto_arima_f(
+                y, 
+                xreg=X,
+                period=self.season_length, 
+                approximation=self.approximation,
+                allowmean=False, allowdrift=False #not implemented yet
+            )
+        return self
+    
+    def predict(
+            self, 
+            h: int, # forecasting horizon 
+            X: np.ndarray = None, # exogenous regressors
+            level: Optional[Tuple[int]] = None, # level 
+        ):
+        fcst = forecast_arima(self.model_, h=h, xreg=X, level=level)
+        mean = fcst['mean']
+        if level is None:
+            return {'mean': mean}
+        return {
+            'mean': mean,
+            **{f'lo-{l}': fcst['lower'][f'{l}%'] for l in reversed(level)},
```

Comment:
Is this compatible with the solution to the reversed levels: https://github.com/Nixtla/statsforecast/issues/150?